staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -99,10 +99,13 @@ func init() {
 }
 
 // Decode attempts to convert the provided data into YAML or JSON, extract the stored schema kind, apply the provided default gvk, and then
-// load that data into an object matching the desired schema kind or the provided into. If into is *runtime.Unknown, the raw data will be
-// extracted and no decoding will be performed. If into is not registered with the typer, then the object will be straight decoded using
-// normal JSON/YAML unmarshalling. If into is provided and the original data is not fully qualified with kind/version/group, the type of
-// the into will be used to alter the returned gvk. On success or most errors, the method will return the calculated schema kind.
+// load that data into an object matching the desired schema kind or the provided into.
+// If into is *runtime.Unknown, the raw data will be extracted and no decoding will be performed.
+// If into is not registered with the typer, then the object will be straight decoded using normal JSON/YAML unmarshalling.
+// If into is provided and the original data is not fully qualified with kind/version/group, the type of the into will be used to alter the returned gvk.
+// If into is nil or data's gvk different from into's gvk, it will generate a new Object with ObjectCreater.New(gvk)
+// On success or most errors, the method will return the calculated schema kind.
+// The gvk calculate priority will be originalData > default gvk > into
```

Comment:
was the comment out of date or why was text dropped?

Reply:
green line 102-105 && 107 is exactly same as previous except the indent format I changed each  `into` situation occupy one independent line.

106 && 108 is two new lines I added, just some complementary information of another `into` situation and schema kind calculated strategy, not sure those two lines is appropriate or not, if not I could delete those.